azurerm/internal/services/network/tests/virtual_hub_route_table_test.go

```diff
@@ -0,0 +1,250 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMVirtualHubRouteTable_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_route_table", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubRouteTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMVirtualHubRouteTable_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_route_table", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubRouteTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMVirtualHubRouteTable_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMVirtualHubRouteTable_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_route_table", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubRouteTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMVirtualHubRouteTable_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_virtual_hub_route_table", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMVirtualHubRouteTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMVirtualHubRouteTable_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualHubRouteTableExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
```

Comment:
could we add another step going back to basic here?

Reply:
updated